savaslabs/savaslabs.github.io
# Issues #4678 and #4679: messages and comment count

Fixes issues [#4678](https://pm.savaslabs.com/issues/4678) and [#4679](https://pm.savaslabs.com/issues/4679)

## Summary of changes

1. Adds the comment count link to the top of blog posts that have 1 or more comments
2. Adds success and error messages to the comment form
3. Cleans up the main.js file and updates comment app code for coding standards

## Notes

There are a couple of things I didn't recreate:

- Comment count in the "Comments" header. We can recreate this if we want to include it in the redesigned comments section, but I didn't want to deal with the refactoring it would take at this point (I can explain more if needed)
- I didn't resolve the issue of fixing the success message on a second comment submission. We could just remove the "leave a comment" link and force the user to refresh the page to submit another comment (which is, as we discussed, an edge case anyway). Thoughts?
- Error message right above the offending field

~~Once we resolve these two issues, the last thing i need to do before we can deploy this is using the comment server variable in `api.js` instead of the hardcoded value.~~ This is done.

## To test

- Pull this code and start up your local comments server
- `gulp clean` + `gulp serve`
- Leave some comments and ensure the comment count link at the top of the post is accurate and links to the comment form
- Make sure posts with no comments do not have the comment count link
- Try submitting a comment with the wrong nocaptcha answer. You should get an error message at the top of the form, and the nocaptcha field should have a red border.
- When you submit a comment, the comment should appear, and you should see a success message that fades after 10 seconds

### Pull request reviewer

As the reviewer, I have verified the following:
